### PR TITLE
Create CODE_OF_CONDUCT.md

### PR DESCRIPTION
**Pull request by ZyanKLee**
ideas for changes to this CoC are welcome.

### CODE REVIEW

**Comment by TheDukeofErl** [comment]
I'm going to start with stating that I recognize that CoCs are, in general, a relatively touchy and divisive subject. 

If we feel the need to have a lengthier and more complex CoC, I'm partial to the FreeBSD one: https://www.freebsd.org/internal/code-of-conduct/. In my opinion, the language used in it feels less like a legal document and comes across as more easily readable. It also focuses more on positive behavior rather than negative behavior and punishments. I view these as a positive on the community. I would prefer this CoC to the currently proposed one.

That said, the community here is rather small yet, and a simpler CoC may work better, such as the Ruby one: https://www.ruby-lang.org/en/conduct/. This one is very brief, which lends greatly to being readable, though it's length may also be viewed as a negative.

Another option would be to maintain the status quo: leaving the repo sans CoC. I'm sure that there would be mixed feelings across the board with this option and I'm not sure it's the best choice given the recent transition to an organization.

There's also the option of making our own, unique CoC, based off others, though this seems like it has been a pitfall for many in the past.

In general, with a CoC I err on the side of caution and would definitely like to propose that this is left up for a while as an issue so that people can respond (and hopefully feel comfortable in doing so).

**Comment by AXDOOMER** [comment]
I hate code of conducts. I think they are demeaning as they assume people can't act like adults. I personally think adding "Please be nice to each others" to the main README file would be more than enough, but even then, people should already be aware of what's reasonable or not to do on a professional forum like Github.

**Comment by ZyanKLee** [comment]
I actually have just taken the text suggested by GitHub, read it and found nothing in it, that I would have changed.
Though after reading both of the suggestions by @TheDukeofErl, I have to agree: the ruby version (or a derivative of it) might be a better fitting text.


While I'm also not a fan of telling people how they should behave, in my opinion history has shown that people *have to be told* how to behave in a certain group.

Behavior that is acceptable in a golf club may not be acceptable in a motorcycle club - and vice versa. And both will most likely not be fine with certain behavior that is regularly shown in twitch chats for example.
So before entering either (and also any other community) one should inform themselves on how to act, what to do, what not to do. One source for doing that could be a CoC - and it's a quite obvious source for that.

I do not hate CoCs, but the fact that "common sense" is so uncommon, or at least diverting between people, that I felt the need for suggesting a CoC.

**Comment by AXDOOMER** [comment]
I think lengthy CoCs are a pitfall, as people (trolls and ill-intentioned) will start poking holes in them. See [this](https://github.com/noisetorch/NoiseTorch/pull/288/files#diff-ffdbe3a1e7ee93cacfc080b6c635ccf3a8f6b0f00f2fb884f78c6b5f9dac8fd2) CoC that was suggested by Github. It mentions about 15 characteristics that people can have in this harassment-free community (age, handicap, etc.), but makes no mention of nationality, ability or native language unlike the [Mozilla one](https://www.mozilla.org/en-US/about/governance/policies/participation/). What about political affiliations? Neither mentions that. Does either community endorses harassment based on political affiliation?

I like Ruby's CoC. It short and simple to read. It's not complex and covers many things in less words. Although, it's been subject to debate. Some people argued that the clause "Participants will be tolerant of opposing views." would mean people should tolerate sexism ([link](https://github.com/ruby/www.ruby-lang.org/pull/2690#issuecomment-930575204).) Somebody even said "When interpreting the words and actions of others, participants should always assume good intentions." came from a place of privilege ([link](https://web.archive.org/web/20211001004040/https://twitter.com/JakeHerrington/status/1443286841533374469).)

Whatever the CoC is, there will always be weaknesses. Having a CoC assumes malicious people will read it and suddenly act with appropriate behavior, which is untrue. It will keep having to evolve, which means energy will be spent (read _wasted_) debating and updating it. This is why, in my opinion, it's not worth it. Especially for small organisations where dealing with processes and policies is too much of a burden (if you to have a consistant approach when enforcing your CoC.)

CoCs are a difficult topic. I don't like wasting my time arguing about unproductive things online, which is why I don't like CoCs. It gets into feelings and language semantics very fast when it attracts too much attention. I prefer to withdraw myself from heated discussions as they often give of a negative vibe. I find the use of the temporary ban and permanent ban always justifiable, regardless if there's a CoC or not, to prevent people from engaging with each other when they start being offensive or harass. 

tl;dr a CoC will not prevent unwanted behaviors and bans are enough to send a message. Issues can be dealt with on a case-by-case basis and have consequences for the perpetrators even without a CoC, thus CoCs are merely virtue signalling as they achieve nothing except make people feel good and safer.
